gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/GfxdTXStateProxy.java

```diff
@@ -426,6 +426,13 @@ final void cleanupIndexEntryForDestroy(Object container,
                 oldRowLocation /* replacement */, false, null,
                 false /* isPutDML */);
 
+        // For SNAP-2620. Better would be to clean the wrapper itself when such scenario arises
+        // Check SNAP-2620 commented out code in SortedMao2IndexInsertOperation
+        if (!deleted && indexContainer.isUniqueIndex()) {
+          deleted = SortedMap2IndexInsertOperation.doMe(null, null, indexContainer,
+              indexKey, (RowLocation) wrapper.getUnderlyingRegionEntry(), true,
+              null, false);
+        }
```

Comment:
Replacing during insert does not work for some reason? This looks like a hackish way.